pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -2564,9 +2564,9 @@ pub fn parse_remote_index_path(path: RemotePath) -> Option<Generation> {
 }
 
 /// Given the key of a tenant manifest, parse out the generation number
-pub(crate) fn parse_remote_tenant_manifest_path(path: RemotePath) -> Option<Generation> {
+pub fn parse_remote_tenant_manifest_path(path: RemotePath) -> Option<Generation> {
     static RE: OnceLock<Regex> = OnceLock::new();
-    let re = RE.get_or_init(|| Regex::new(r".+tenant-manifest-([0-9a-f]{8}).json").unwrap());
+    let re = RE.get_or_init(|| Regex::new(r".*tenant-manifest-([0-9a-f]{8}).json").unwrap());
```

Comment:
This method is called sometimes with a full key, sometimes with just the basename.
What prompted this `.+` => `.*` change?
If we're drive-by-fixing stuff anyway, can we also ensure we always evaluate the regex against a basename?

Reply:
> This method is called sometimes with a full key, sometimes with just the basename.
> What prompted this .+ => .* change?

That's why i did the `.+` => `.*` change.